Nio/Authentication/LoginView.swift

```diff
@@ -36,6 +37,23 @@ struct LoginContainerView: View {
 
         store.login(username: username, password: password, homeserver: homeserverURL)
     }
+    
+    private func guessHomeserverURL() {
+        if !username.isEmpty && homeserver.isEmpty {
+            let userparts = username.split(separator: ":")
+            if userparts.count < 2 {
+                return
+            }
+            let mxautodiscovery = MXAutoDiscovery.init(domain: String(userparts[1]))
```

Comment:
Skipping the `String` init thanks to `components(separatedBy:)` and we don't have to explicitly call `MXAutoDiscovery`'s initializer.
```suggestion
            let mxautodiscovery = MXAutoDiscovery(domain: userparts[1])
```